### DISCUSSION TITLE
Variant Specific Information

### DISCUSSION THREAD

**Discussion by cancan101** [pimcore/pimcore#16944] [Q&A / Support] [answered]
If I am using Variants (or "Data Inheritance" more generally) is there anyway to add additional fields in the child / Variant? For example I have location specific variants of an Object. I want to inherit all of the fields but the child / Variant also need a field to track its "location". The "location" field does not belong on / make sense at the parent level (nor would I want to inherit its value). 

This is related to #16941 (CC @dpfaffenbauer)

**Comment by fashxp** [comment, marked as answer]
you can use object bricks for that

**Reply by cancan101** [reply]
Is there anything built in to Pimcore to add / remove an Object Brick on an object based on another attribute? Or is the expectation the User exclusively uses the UI themselves to add / remove the Brick? 

For example in my case I would want to enforce that all variants have the Brick added. I can write some custom logic to do that, but wanted to see if there was anything built in.

**Reply by markus-moser** [reply]
There is no built in option. You would need a event listener for this.
